portfolio/src/main/webapp/js/custom-script.js

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.let map;
+/* exported onLoad */
+/* exported codeAddress */
+/* exported marker */
+// Neccessary constants or else variables will return as
+// 'undefined' in lint checks
+let map;
```

Comment:
What is the reason of declaring these variables "map, geocoder, marker" as global scope? Usually we would try to limit the scope of variable unless it's necessary?

Reply:
looks like all these map related functions can be grouped into a class?

Reply:
I declared them as global variables so that the other helper methods (for example, toggleBounce() that I have added in the next PR) can access them. What would the benefit of grouping them into a class be?

Reply:
Global variables can become hard to manage if many different files/functions are modifying them. They also pollute the global namespace (maybe an external library creates a global map variable or function too).

There's nothing wrong with globals (there are certainly appropriate times for them), but like Tong said, it's nice to limit the scope of your variables to where they're needed so you avoid having weird bugs down the line.

Reply:
I was not able to decrease the scope of map, marker, and geocode in this pull request. I outlined the problems I faced and what I'll be doing next week to decrease the scope in an FYI below.